pkg/apiserver/clusterinfo/fetch.go

```diff
@@ -0,0 +1,227 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusterinfo
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/pingcap/kvproto/pkg/metapb"
+
+	"github.com/pingcap/log"
+
+	"github.com/pkg/errors"
+
+	"github.com/pingcap-incubator/tidb-dashboard/pkg/utils/clusterinfo"
+)
+
+// fetcher is an interface for concurrently fetch data and store it in `info`.
+type fetcher interface {
+	// fetch fetches the data, and if any unrecoverable error exists, it will return error.
+	fetch(ctx context.Context, info *ClusterInfo, service *Service) error
+}
+
+// etcdFetcher fetches etcd, and parses the ns below:
+// * /topology/grafana
+// * /topology/alertmanager
+// * /topology/tidb
+type etcdFetcher struct{}
+
+func (f etcdFetcher) fetch(ctx context.Context, info *ClusterInfo, service *Service) error {
```

Comment:
So why not simply call it "fetchPD", "fetchTiDB", etc?

Reply:
I change it to tidbFetcher.